provider/pkg/autonaming/application_test.go

```diff
@@ -73,7 +74,123 @@ func Test_getDefaultName(t *testing.T) {
 				MinLength: tt.minLength,
 				MaxLength: tt.maxLength,
 			}
-			got, err := getDefaultName(nil, urn, autoNamingSpec, tt.olds, tt.news)
+			got, err := getDefaultName(nil, urn, autoNamingSpec, tt.olds, tt.news, nil)
+			if tt.err != nil {
+				require.EqualError(t, err, tt.err.Error())
+				return
+			} else {
+				require.NoError(t, err)
+			}
+			if !tt.comparison(got) {
+				t.Errorf("getDefaultName() = %v for spec: %+v", got, autoNamingSpec)
+			}
+			t.Logf("getDefaultName() = %v for spec: %+v", got, autoNamingSpec)
+		})
+	}
+}
+
+func Test_getDefaultName_withAutoNameConfig(t *testing.T) {
+	const sdkName = "autoName"
+	tests := []struct {
+		name           string
+		resourceName   string
+		autoNameConfig AutoNamingConfig
+		minLength      int
+		maxLength      int
+		olds           resource.PropertyMap
+		news           resource.PropertyMap
+		err            error
+		comparison     func(actual resource.PropertyValue) bool
+	}{
+		{
+			name:         "Name specified explicitly",
+			resourceName: "myName",
+			news: resource.PropertyMap{
+				resource.PropertyKey(sdkName): resource.NewStringProperty("newName"),
+			},
+			autoNameConfig: AutoNamingConfig{
+				AutoTrim:              true,
+				RandomSuffixMinLength: 1,
+			},
+			maxLength:  2,
+			comparison: equals(resource.NewStringProperty("newName")),
+		},
+		{
+			name:         "Use old name",
+			resourceName: "myName",
+			olds: resource.PropertyMap{
+				resource.PropertyKey(sdkName): resource.NewStringProperty("oldName"),
+			},
+			autoNameConfig: AutoNamingConfig{
+				AutoTrim:              true,
+				RandomSuffixMinLength: 1,
+			},
+			maxLength:  2,
+			comparison: equals(resource.NewStringProperty("oldName")),
+		},
+		{
+			resourceName: "myReallyLongName",
+			name:         "Autoname with constraints on max length",
+			maxLength:    12,
+			autoNameConfig: AutoNamingConfig{
+				AutoTrim: true,
+			},
+			comparison: within(12, 12),
```

Comment:
Any chance we can see `equals(resource.NewStringProperty(".."))` here or it's omitted because it is random?

Reply:
I think I might be able to do a regex check because only the end should be random.

Reply:
Updated